Lib/unittest/test/testmock/testasync.py

```diff
@@ -610,76 +614,92 @@ def test_assert_awaited_but_not_called(self):
             self.mock.assert_called()
 
     def test_assert_has_calls_not_awaits(self):
-        kalls = [call('NormalFoo')]
+        kalls = [call('foo')]
         with self.assertWarns(RuntimeWarning):
-            self.mock('NormalFoo')
+            # Will raise a warning because never awaited
+            self.mock('foo')
         self.mock.assert_has_calls(kalls)
         with self.assertRaises(AssertionError):
             self.mock.assert_has_awaits(kalls)
 
-    def test_assert_has_mock_calls_on_async_mock(self):
-        mock = AsyncMock()
+    def test_assert_has_mock_calls_on_async_mock_no_spec(self):
         with self.assertWarns(RuntimeWarning):
-            mock()
+            # Will raise a warning because never awaited
+            self.mock()
         kalls_empty = [('', (), {})]
-        self.assertEqual(mock.mock_calls, kalls_empty)
+        self.assertEqual(self.mock.mock_calls, kalls_empty)
 
-        kalls_args = ([call('NormalFoo'), call('baz')])
         with self.assertWarns(RuntimeWarning):
-            self.mock('NormalFoo')
+            # Will raise a warning because never awaited
+            self.mock('foo')
             self.mock('baz')
+        kalls_args = ([call(), call('foo'), call('baz')])
         self.assertEqual(self.mock.mock_calls, kalls_args)
 
-        a_mock = AsyncMock(AsyncClass)
+    def test_assert_has_mock_calls_on_async_mock_with_spec(self):
+        a_class_mock = AsyncMock(AsyncClass)
         with self.assertWarns(RuntimeWarning):
-            a_mock.async_method()
-        self.assertEqual(a_mock.async_method.mock_calls, kalls_empty)
-        self.assertEqual(a_mock.mock_calls, [call.async_method()])
+            # Will raise a warning because never awaited
+            a_class_mock.async_method()
+        kalls_empty = [('', (), {})]
+        self.assertEqual(a_class_mock.async_method.mock_calls, kalls_empty)
+        self.assertEqual(a_class_mock.mock_calls, [call.async_method()])
 
-        kalls_args_kwargs = [call(), call(1, 2, 3, a=4, b=5)]
         with self.assertWarns(RuntimeWarning):
-            a_mock.async_method(1, 2, 3, a=4, b=5)
-        self.assertEqual(a_mock.async_method.mock_calls, kalls_args_kwargs)
+            # Will raise a warning because never awaited
+            a_class_mock.async_method(1, 2, 3, a=4, b=5)
+        kalls_args_kwargs = [call(), call(1, 2, 3, a=4, b=5)]
+        self.assertEqual(a_class_mock.async_method.mock_calls, kalls_args_kwargs)
+        self.assertEqual(
+            a_class_mock.mock_calls,
+            [
+                call.async_method(),
+                call.async_method(1, 2, 3, a=4, b=5)
+            ]
+        )
 
     def test_async_method_calls_recorded(self):
         with self.assertWarns(RuntimeWarning):
+            # Will raise a warning because never awaited
```

Comment:
Nitpick: `will raise *warnings* because never awaited, since here there are two calls, and presumably two warnings raised.

The same apply to the same comment below.